mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -1999,6 +2041,9 @@ OpFoldResult ExtractOp::fold(FoldAdaptor) {
     return val;
   if (auto val = foldScalarExtractFromFromElements(*this))
     return val;
+  SmallVector<Value> operands = {getVector()};
+  if (succeeded(foldConstantOp(*this, operands)))
```

Comment:
nit: Could you make the API similar to the other `fold*` methods above? That is, return a `Value`, not a `LogicalResult`?

Reply:
This is a great learning opportunity, sure, thank you.I've finished fold extractOp and insertOp, I'll commit it first, and I'll add fold poison later (I need a little time.).